Changing names of functions in PersistentFilters

katelynienaber opened the pull request:
Signed-off-by: Katelyn Nienaber <katelyn.nienaber@broadcom.com>

## Proposed changes

Changing the names of the functions in PersistentFilters in a separate PR, as requested by Vit on [this PR](https://github.com/zowe/vscode-extension-for-zowe/pull/920).

I changed the names because having "history" and "recall" arrays in the settings file is not clear. So "history" is changed to `searchHistory` and "recall" is changed to `fileHistory` because those are the types of items stored in each array...search strings and file paths.

Also I reordered the functions in the file so that similar functions for each persistent array are grouped together (ie. getSearchHistory grouped with getFileHistory/getFavorites, etc.)

## Release Notes
<!-- Include the Milestone Number and a small description of your change that will be added to the changelog -->
<!-- If there is a linked issue, it should have the same milestone as this PR -->

Milestone: 1.8
Changelog: Changed names of history and recall functions in PersistentFilters to searchHistory and fileHistory.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] `npm run vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

VitGottwald commented on __tests__/__integration__/DatasetTree.integration.test.ts:
```diff
@@ -236,15 +236,15 @@ describe("DatasetTree Integration Tests", async () => {
-        const recall = testTree.getRecall();
+    it("Tests that addFileHistory adds a recently-opened file to the list", async () => {
+        testTree.addFileHistory(`[${sessNode.getLabel()}]: ${pattern}.EXT.SAMPLE.PDS(TESTMEMB)`);
+        const recall = testTree.getFileHistory();
```
How about renaming `recall` to `lastDataset` ?

katelynienaber replied:
Eeeee but right now all the `FileHistory` functions have matching names...and it would be nice if it had the format of `sometypeofHistory()` because then it will be matching with the other history function...addSearchHistory, getSearchHistory, etc.

Why do you want to rename it? Maybe we can find a better name that still matches with this format?

VitGottwald replied:
The reason why I suggested to rename is that identifiers should describe the meaning of the variable, function ... and `recall` does give me any idea what it is about.

VitGottwald replied:
Maybe there is a misunderstanding, here, I was referring to the variable name, not to the function name.

katelynienaber replied:
I think I just failed to remove all the old names from the test files

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /**
+     * Returns the current contents of the persistent favorites array
+     *
+     * @returns {string[]} persistent array of favorited data sets, USS files, and jobs
```
We should not duplicate the type information in JSDoc.

```suggestion
     * @returns persistent array of favorited data sets, USS files, and jobs
```

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /**
+     * Returns the current contents of the persistent recently-opened file history array
+     *
+     * @returns {string[]} persistent array of recently-opened files
```
We should not duplicate the type information in JSDoc.
```suggestion
     * @returns persistent array of recently-opened files
```

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /**
+     * Returns the current contents of the persistent recently-opened file history array
+     *
+     * @returns {string[]} persistent array of recently-opened files
+     */
+    public getFileHistory() {
+        return this.mFileHistory;
+    }
```
```suggestion
     * @returns persistent array of recently-opened files
     */
    public getFileHistory(): string[] {
        return this.mFileHistory;
    }
```

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /**
+     * Returns the current contents of the persistent search history array
+     *
+     * @returns {string[]} persistent array of search history items
+     */
+    public getSearchHistory() {
+        return this.mSearchHistory;
+    }
```
```suggestion
     * @returns persistent array of search history items
     */
    public getSearchHistory(): string[] {
        return this.mSearchHistory;
    }
```

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /**
+     * Returns the current contents of the persistent sessions array
+     *
+     * @returns {string[]} persistent array of sessions and their settings
+     */
+    public getSessions() {
+        return this.mSessions;
+    }
```
```suggestion
     * @returns persistent array of sessions and their settings
     */
    public getSessions(): string[] {
        return this.mSessions;
    }
```

VitGottwald commented:
Thank you for extracting the rename to a separate PR. This is very helpful!

VitGottwald commented:
I only noticed some duplicated types in code and JSDoc and added suggestions to remove types form JSDoc. I know they come from the old code, but I think it would be good to do this small cleanup. Other than that, it is good to go! Hope Jenkins starts working soon :-/

katelynienaber commented:
> I only noticed some duplicated types in code and JSDoc and added suggestions to remove types form JSDoc. I know they come from the old code, but I think it would be good to do this small cleanup. Other than that, it is good to go! Hope Jenkins starts working soon 🤞

Those have been fixed :) GitHub has this cool new `Commit suggestion` button (or maybe it is not new and I am blind XD)

Added a comment to your first bit ^^

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+    /* Update functions, for updating the settings.json file in VSCode
+    /*********************************************************************************************************************************************/
+
+    /**
+     * Updates the stored favorites array in settings.json
+     */
```
I think the block comment above + function name makes it clear and this comment is redundant.

VitGottwald replied:
Looking at this again, alternatively removing the block comment is a better solution since this jsdoc shows as a hint in intellisense.

katelynienaber replied:
I would like to leave the block comment, as it helps organize the file. Each set of functions in the file under a block comment have similar functions. So when you are scrolling around you can quickly get to the section you are looking for.

The Intellisense thing is also useful.

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
     }
 
-    private async updateRecall() {
+    /**
+     * Updates the search history array in settings.json
+     */
+    private async updateSearchHistory() {
```
I think the block comment above + function name makes it clear and this comment is redundant.

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
         }
     }
 
+    /**
+     * Updates the stored sessions array in settings.json
+     */
```
I think the block comment above + function name makes it clear and this comment is redundant.

VitGottwald commented on src/PersistentFilters.ts:
```diff
@@ -230,4 +278,45 @@ export class PersistentFilters {
         }
     }
+
+    /**
+     * Updates the recently-opened file history array in settings.json
+     */
```
I think the block comment above + function name makes it clear and this comment is redundant.

VitGottwald commented:
Thank you for verifying that @zFernand0 !

zFernand0 commented:
Merging as this does not present any conflicts with the webpack PR

VitGottwald commented:
Thank you for merging this @zFernand0 ! You are our hero 💪
